File: 2021/07/05/main.go
package main

func matrixReshape(mat [][]int, r int, c int) [][]int {
	n, m := len(mat), len(mat[0])
	if n*m != r*c {
		return mat
	}
	cnt, res := 0, make([][]int, r)
	for i := range res {
		res[i] = make([]int, c)
		for j := range res[i] {
			res[i][j] = mat[cnt/m][cnt%m]
			cnt++
		}
	}
	return res
}

File: 2021/07/05/main.py
class Solution:
    def matrixReshape(self, mat: List[List[int]], r: int, c: int) -> List[List[int]]:
        if sum([len(m) for m in mat]) != (r*c):
            return mat
        d = len(mat[0])
        m = [[0] * c for j in range(r)]
        for i in range(r * c):
            m[i//c][i%c] = mat[i//d][i%d]
        return m